### PR TITLE
Micro Frontends Using Module Federation In Solid.js

### LINKED ISSUE #6547: Micro Frontends Using Module Federation In Solid.js
#### Micro frontends Using Module Federation In Solid.js
Your title should be descriptive of the article/tutorial. Be Specific. 

Use keyword research to improve your article's SEO.

#### Proposed article introduction
**Micro-frontends allow you to develop a website or web app as a collection of elements that you can create and work on independently. In this tutorial, we'll learn about micro frontend, module federation, and how to use Module Federation to create a micro frontend in Solid.js. We'll also learn how to create and share a Micro-Frontend between Solid.JS and React.**

For example, if the article is based on machine learning, use the following: [Machine learning] Introduction to Machine Learning.
If the article is based on developing an Android application, use the following: [Android] Developing Apps using Android.

ONLY viable topics to pick from:
- API
- Artificial Intelligence
- Blockchain
- Containers
- Edge Computing
- Languages
- Machine Learning
- Networking
- Node.js
- Security

#### Key takeaways 
At the end of this tutorial, the readers would be able to
- Know what micro frontend is.
- How it works.
- What problem does it solve?
- create a host app
- create a remote app
- consume micro frontend.

#### Article quality
This article tends to show how flexible micro frontend is when it comes to cross-platforming as I'll also be showing how to share between different frameworks using React and Solid.js as an example. Most articles cover using micro frontend in React, angular, and others but not on Solid.js this articles cover that.

### PR DESCRIPTION
**Pull request by oyedeletemitope**
## Our **Peer Reviewer and Content Moderator teams do NOT provide any revisions services.** All revisions and edits should be completed on your own forked repo (as to not take up room in the queue) in order for our team to review them all in a timely manner. 

**ANY ARTICLE SUBMITTED WITH GLARING ERRORS WILL BE IMMEDIATELY CLOSED.**

>As a rule of thumb - please be sure to only submit articles (pull requests) that are fully polished and ready to be published. Be sure to go through our resources documents for extra and 3rd party (vetted) resources to help improve overall technical writing.

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x ] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

`This closes #6547 

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

File: content/articles/microfrontend-using-module-federation-in-solid.js/index.md [hunk 1]
```diff
@@ -0,0 +1,292 @@
+
+There’s a lot of buzz going on around micro frontends right now. You may be asking yourself this question, how do I try it out? Not to worry. In this tutorial, we’ll be discussing micro frontends. We'll also set up micro frontends using module federation in Solid.js. Also, we take a look at how to share a micro frontend between Solid.js and React using module federation. So, let’s jump right in!
+
+### What Are Micro Frontends?
+
+Micro frontend originated from the concept of microservice. When we’re talking about micro frontends, think of it as a website composed of different web pages. These pages consist of features created by independent teams. Let’s look at Netflix for example. Imagine the header section of Netflix, the header video, and the carousel built by a separate team. Also, built by a different team is the video section. These teams have their different ways of working and they build up the main Netflix website. Another way to think of micro frontend could be separate pages on the main site. These pages combined together build the website.
```
Comment (jumamiller) [inline content/articles/microfrontend-using-module-federation-in-solid.js/index.md:6]: Let's ensure that a paragraph is as short and precise. This looks too wordy. Update where applicable!

File: content/articles/microfrontend-using-module-federation-in-solid.js/index.md [hunk 2]
```diff
@@ -0,0 +1,292 @@
...
+
+Micro-frontends are one popular approach to increasing flexibility. It allows teams to combine components built in different frameworks or libraries. There are several ways to put in place micro frontends. We’ll be focusing on module federation in this tutorial but first, let’s look at the benefits of micro frontend and how we split our apps.
+
+### Benefits Of Micro Frontends
+
```
Comment (jumamiller) [inline content/articles/microfrontend-using-module-federation-in-solid.js/index.md:11]: Do away with these spaces throughout your article!

File: content/articles/microfrontend-using-module-federation-in-solid.js/index.md [hunk 3]
```diff
@@ -0,0 +1,292 @@
...
+### How Do We Split Apps?
+
+Here are some examples of how developers split large apps:
+
+#### By Page
+
+Sometimes, having many pages opened at the same time can crash older devices. In such cases, splitting by page is the safest option. If you have good routing, you can run separate, specific micro-apps for each page. This is also beneficial to your team's developers.
+
+#### By Functionality
```
Comment (jumamiller) [inline content/articles/microfrontend-using-module-federation-in-solid.js/index.md:28]: Follow the English rules. No uppercase in the middle of the sentence? Update where applicable!

**Review by jumamiller — CHANGES_REQUESTED** [review]
What are the requisites to follow along with this tutorial? if not available, why YARN and not the popular NPM?

**Comment by jumamiller** [comment]
ALso ensure your code snippets are well explained!

**Comment by ahmadmardeni1** [comment]
Plagiarism check done ✅

**Comment by oyedeletemitope** [comment]
> 
I'm using yarn because npm has been giving some errors unlike yarn plus it's what I'm comfortable with.

**Comment by oyedeletemitope** [comment]
> What are the requisites to follow along with this tutorial? if not available, why YARN and not the popular NPM?

I'm using yarn because npm has been giving some errors unlike yarn plus it's what I'm comfortable with.

**Comment by oyedeletemitope** [comment]
> ALso ensure your code snippets are well explained!

changes made. please let me know if there's anything else

**Comment by jumamiller** [comment]
> > What are the requisites to follow along with this tutorial? if not available, why YARN and not the popular NPM?
> 
> I'm using yarn because npm has been giving some errors unlike yarn plus it's what I'm comfortable with.

Remember an article is for all readers interested in your topic and should be as general as possible! Just give alternative commands in NPM.

**Comment by oyedeletemitope** [comment]
> > > What are the requisites to follow along with this tutorial? if not available, why YARN and not the popular NPM?
> > 
> > 
> > I'm using yarn because npm has been giving some errors unlike yarn plus it's what I'm comfortable with.
> 
> Remember an article is for all readers interested in your topic and should be as general as possible! Just give alternative commands in NPM.

I have added the npm commands as requested

**Comment by jumamiller** [comment]
Hi, could you also work on the repetitions? For instance, you're using one word a couple of times in a sentence,`In this tutorial, we'll be discussing micro frontends. We'll also set up micro frontends using module federation in Solid.js. `

In this case, `micro frontends` has been used twice. This is not the only case in your article, others such as `that` among others are very repetitive. Kindly be precise and and avoid repeating yourself.

**Comment by jumamiller** [comment]
> Hi, could you also work on the repetitions? For instance, you're using one word a couple of times in a sentence,`In this tutorial, we'll be discussing micro frontends. We'll also set up micro frontends using module federation in Solid.js. `
> 
> In this case, `micro frontends` has been used twice. This is not the only case in your article, others such as `that` among others are very repetitive. Kindly be precise to avoid repeating yourself.

Ensure images extensions are also in lowercase, let's just maintain the standards put in place!, I. e `image.png` and not `image.PNG`.

**Comment by oyedeletemitope** [comment]
> > Hi, could you also work on the repetitions? For instance, you're using one word a couple of times in a sentence,`In this tutorial, we'll be discussing micro frontends. We'll also set up micro frontends using module federation in Solid.js. `
> > In this case, `micro frontends` has been used twice. This is not the only case in your article, others such as `that` among others are very repetitive. Kindly be precise to avoid repeating yourself.
> 
> Ensure images extensions are also in lowercase, let's just maintain the standards put in place!, I. e `image.png` and not `image.PNG`.

HI @jumamiller Changes made as requested

**Review by jumamiller — APPROVED** [review]
Hi, @oyedeletemitope Kindly review if your details are well captured and that your article is well-edited to deliver the concept as you expected. Thank you!

**Comment by oyedeletemitope** [comment]
Thanks @jumamiller . I've gone through it. Your edits and suggestions were invaluable. Thank you so much

**Comment by LinusMuema** [comment]
Hi @oyedeletemitope 

Good job on the article. Please check my commit for the changes.

cc: @jumamiller
